src/vs/workbench/contrib/terminal/browser/terminalInstance.ts

```diff
@@ -1058,6 +1060,47 @@ export class TerminalInstance extends Disposable implements ITerminalInstance {
 		this._terminalAltBufferActiveContextKey.set(!!(this.xterm && this.xterm.raw.buffer.active === this.xterm.raw.buffer.alternate));
 	}
 
+	private async _shouldPasteText(text: string): Promise<boolean> {
+		const textForLines = text.split(/(\r?\n)/);
+		let confirmation: IConfirmationResult;
+
+		// If the clipboard has only one line, no prompt will be triggered
+		if (textForLines.length === 1 || !this._configurationService.getValue<boolean>(TerminalSettingId.EnableMultiLinePasteWarning)) {
+			return true;
+		} else {
+			const message = nls.localize('confirmMoveTrashMessageFilesAndDirectories', "Are you sure you want to paste the following {0} lines to the terminal?", text.split(/\r\n|\n|\r/).length);
+
+			const displayItemsCount = 3;
+			const ellipsis = '…';
+
+			let detail = textForLines
+				.slice(0, displayItemsCount)
+				.map(s => s.length >= 30 ? s.slice(0, 30) + ellipsis : s)
+				.join('');
```

Comment:
Think we need to join with \n?

Reply:
> Think we need to join with \n?

I think we don’t need join with \n, Because  we split ’textForLines‘ is included \n, we show the first three parts, Ideally it contains two lines of text and a \n

Reply:
Ah ok, reading that part of the code was a little confusing.